internal/auth0/organization/resource_member_roles.go

```diff
@@ -0,0 +1,140 @@
+package organization
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/auth0/go-auth0/management"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/auth0/terraform-provider-auth0/internal/config"
+	internalSchema "github.com/auth0/terraform-provider-auth0/internal/schema"
+)
+
+// NewMemberRolesResource will return a new auth0_organization_member_roles (1:many) resource.
+func NewMemberRolesResource() *schema.Resource {
+	return &schema.Resource{
+		Description:   "This resource is used to manage the roles assigned to an organization member.",
+		CreateContext: createOrganizationMemberRoles,
+		ReadContext:   readOrganizationMemberRoles,
+		UpdateContext: updateOrganizationMemberRoles,
+		DeleteContext: deleteOrganizationMemberRoles,
+		Importer: &schema.ResourceImporter{
+			StateContext: internalSchema.ImportResourcePairID("organization_id", "user_id"),
+		},
+		Schema: map[string]*schema.Schema{
+			"organization_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the organization.",
+			},
+			"user_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The ID of the user that is an organization member.",
+			},
+			"roles": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Description: "The role ID(s) to assign to the organization member.",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func createOrganizationMemberRoles(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	mutex := meta.(*config.Config).GetMutex()
+
+	organizationID := data.Get("organization_id").(string)
+	userID := data.Get("user_id").(string)
+
+	mutex.Lock(organizationID)
+	defer mutex.Unlock(organizationID)
```

Comment:
Because members are now part of the organization data source, for data consistency it would be better to lock on OrgID only IMO. We could dive deeper into the topic and follow up with another PR in case we need to change this, as it's not just here but within all the resource relationship resources related to orgs.

Reply:
Spinning-out an investigation task to see if/how this should be addressed across all resources. But for now, this appears to function.